File: LanguageTutorials/DesignPatterns/src/main/java/org/bytescale/patterns/decorator/Soy.java
package org.bytescale.patterns.decorator;

/**
 * Created by IntelliJ IDEA.
 * User: jared
 * Date: 6/6/11
 * Time: 7:53 AM
 * To change this template use File | Settings | File Templates.
 */
public class Soy extends CondimentDecorator {
    private Beverage beverage;

    public Soy(Beverage beverage) {
        this.beverage = beverage;
    }

    @Override
    public double cost() {
        return this.beverage.cost() + .15;
    }

    @Override
    public String getDescription() {
        return this.beverage.getDescription() + ", Soy";
    }
}

File: LanguageTutorials/CSDesignPatterns/Org.Bytescale.Patterns/Decorator/Soy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Org.Bytescale.Patterns.Decorator
{
    public class Soy : CondimentDecorator
    {
        private Beverage _beverage;

        public Soy(Beverage beverage) {
            _beverage = beverage;
        }

        public override string Description
        {
            get { return _beverage.Description + ", Soy"; }
        }

        public override double Cost
        {
            get { return _beverage.Cost + .15; }
        }
    }
}
